wolfcrypt/src/aes.c

```
@@ -2838,6 +2838,12 @@ static WARN_UNUSED_RESULT int wc_AesDecrypt(
         word32 localSz = 32;
     #endif
 
+    #ifdef WOLFSSL_MAXQ10XX_CRYPTO
+        if (wc_MAXQ10XX_AesSetKey(aes, userKey, keylen) != 0) {
```

I always prefer to return the error code of the function that failed, rather than converting it to a different one (WC_HW_E in this case). Just makes debugging easier if you know the original return code of the failing function.

---

Convention in this function is to return WC_HW_E . Going to stick with it.